File: api/booking-service/src/main/java/com/carrental/bookingservice/model/dto/BookingCostRequestDTO.java
package com.carrental.bookingservice.model.dto;

import com.fasterxml.jackson.annotation.JsonFormat;
import lombok.Data;

import javax.validation.constraints.NotNull;
import java.time.LocalDate;

@Data
public class BookingCostRequestDTO {

    @NotNull
    private Long vehicleId;

    @NotNull
    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd")
    private LocalDate reservationDate;

    @NotNull
    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd")
    private LocalDate returnDate;
}

File: frontend-app/shared/model/BookingCostRequestDTO.ts
export class BookingCostRequestDTO {
    vehicleId: string;
    reservationDate: string;
    returnDate: string;

    constructor(vehicleId: string, reservationDate: string, returnDate: string) {
        this.vehicleId = vehicleId;
        this.reservationDate = reservationDate;
        this.returnDate = returnDate;
    }
}
